### PR TITLE
Further Catch Up improvements

### PR DESCRIPTION
**Pull request by dmdashenkov**
Now the Datastore-native query is built regarding the event type which is passed in the `EventStreamQuery`.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

File: gcd/src/main/java/org/spine3/server/storage/datastore/DsEventRecordStorage.java [hunk 1]
```diff
@@ -63,31 +68,54 @@ protected DsEventRecordStorage(DsRecordStorage<EventId> storage) {
...
+            for (Filter filter : filters) {
+                final StructuredQuery<Entity> query = entityQuery.setFilter(filter)
+                                                                 .build();
+                result.add(query);
+            }
+            return result;
         }
     }
 
+    private static Iterable<Filter> mergeFilters(Iterable<Filter> typeFilters,
```
Comment (armiol) [inline gcd/src/main/java/org/spine3/server/storage/datastore/DsEventRecordStorage.java]: Please document this.

File: gcd/src/main/java/org/spine3/server/storage/datastore/DsEventRecordStorage.java [hunk 2]
```diff
@@ -63,31 +68,54 @@ protected DsEventRecordStorage(DsRecordStorage<EventId> storage) {
...
+        final Collection<Filter> typeFilters = buildTypeFilters(sourceQuery);
+        if (typeFilters.isEmpty()) {
+            if (timeFilter != null) {
+                entityQuery.setFilter(timeFilter);
+            }
+            return singleton(entityQuery.build());
         } else {
-            return entityQuery.setFilter(filter)
-                              .build();
+            final Collection<StructuredQuery<Entity>> result = new LinkedList<>();
```
Comment (armiol) [inline gcd/src/main/java/org/spine3/server/storage/datastore/DsEventRecordStorage.java]: Please use `newLinkedList()` for consistency with our code style.

Same applies to the similar statement below.

**Review by armiol — APPROVED** [review]
@dmdashenkov LGTM except for minor comments. Please address those and feel free to merge.

**Pull request merged**